cmd/ebrelayer/relayer/relayer.go

```diff
@@ -0,0 +1,119 @@
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/swishlabsco/cosmos-ethereum-bridge/cmd/ebrelayer/contract"
+	"github.com/swishlabsco/cosmos-ethereum-bridge/cmd/ebrelayer/events"
+	"github.com/swishlabsco/cosmos-ethereum-bridge/cmd/ebrelayer/txs"
+)
+
+// -------------------------------------------------------------------------
+// Starts an event listener on a specific network, contract, and event
+// -------------------------------------------------------------------------
+
+func InitRelayer(cdc *amino.Codec, chainId string, provider string,
+	contractAddress common.Address, eventSig string,
+	validatorFrom string) error {
+
+	validatorAccAddress, validatorName, err := sdkContext.GetFromFields(validatorFrom)
+	if err != nil {
+		fmt.Printf("failed to get from fields: %v", err)
+		return err
+	}
+	validatorAddress := sdk.ValAddress(validatorAccAddress)
+
+	passphrase, err := keys.GetPassphrase(validatorFrom)
+	if err != nil {
+		return err
+	}
+
+	//Test passhprase is correct
+	_, err = authtxb.MakeSignature(nil, validatorName, passphrase, authtxb.StdSignMsg{})
+	if err != nil {
+		fmt.Printf("passphrase error: %v", err)
+		return err
+	}
+
+	// Start client with infura ropsten provider
+	client, err := SetupWebsocketEthClient(provider)
+	if err != nil {
+		fmt.Errorf("%s", err)
+	}
+	fmt.Printf("\nStarted ethereum websocket with provider: %s", provider)
+
+	// We need the contract address in bytes[] for the query
+	query := ethereum.FilterQuery{
+		Addresses: []common.Address{contractAddress},
+	}
+
+	// We will check logs for new events
+	logs := make(chan types.Log)
+
+	// Filter by contract and event, write results to logs
+	sub, err := client.SubscribeFilterLogs(context.Background(), query, logs)
+	if err != nil {
+		fmt.Errorf("%s", err)
+	} else {
+		fmt.Printf("\nSubscribed to contract events on address: %s\n", contractAddress.Hex())
+	}
+
+	// Load Peggy Contract's ABI
+	contractABI := contract.LoadABI()
+
+	for {
+		select {
+		// Handle any errors
+		case err := <-sub.Err():
+			log.Fatal(err)
+		// vLog is raw event data
+		case vLog := <-logs:
+			// Check if the event is a 'LogLock' event
+			if vLog.Topics[0].Hex() == eventSig {
+				fmt.Printf("\n\nNew Lock Transaction:\nTx hash: %v\nBlock number: %v",
+					vLog.TxHash.Hex(), vLog.BlockNumber)
+
+				// Parse the event data into a new LockEvent using the contract's ABI
+				event := events.NewLockEvent(contractABI, "LogLock", vLog.Data)
+
+				// Add the event to the record
+				events.NewEventWrite(vLog.TxHash.Hex(), event)
+
+				// Parse the event's payload into a struct
+				claim, err := txs.ParsePayload(validatorAddress, &event)
+				if err != nil {
+					fmt.Errorf("Error: %s", err)
+				}
+
+				// Initiate the relay
+				err := txs.RelayEvent(chainId, cdc, validatorAddress, validatorName, passphrase, &claim)
+				if err != nil {
+					fmt.Errorf("Error: %s", err)
```

Comment:
```suggestion
					return err
```

Reply:
fixed in next push

Reply:
https://github.com/cosmos/peggy/commit/102c1283899e9820cd4c2123eb5d2b94b75ad070